common/transport/DcgmIpc.cpp

```diff
@@ -286,8 +288,40 @@ dcgmReturn_t DcgmIpc::InitTCPListenerSocket()
         return DCGM_ST_OK;
     }
 
+    memset(&listenAddr4, 0, sizeof(listenAddr4));
+    listenAddr4.sin_family = AF_INET;
+    listenAddr4.sin_port   = htons(m_tcpParameters.value().port);
+
+    memset(&listenAddr6, 0, sizeof(listenAddr6));
+    listenAddr6.sin6_family = AF_INET6;
+    listenAddr6.sin6_port   = htons(m_tcpParameters.value().port);
+
+    if (m_tcpParameters.value().bindIPAddress.size() > 0)
+    {
+        if (inet_aton(m_tcpParameters.value().bindIPAddress.c_str(), &listenAddr4.sin_addr) == 1)
+        {
+            memcpy(&listenAddr, &listenAddr4, sizeof(listenAddr4));
+        }
+        else if (inet_pton(AF_INET6, m_tcpParameters.value().bindIPAddress.c_str(), &listenAddr6.sin6_addr) == 1)
+        {
+            memcpy(&listenAddr, &listenAddr6, sizeof(listenAddr6));
+        }
+        else
+        {
+            DCGM_LOG_ERROR << "Unable to convert \"" << m_tcpParameters.value().bindIPAddress
+                           << "\" to a network address.";
+            return DCGM_ST_GENERIC_ERROR;
+        }
+    }
+    else
+    {
+        /* Bind to all interfaces (IPv4 and IPv6) */
+        listenAddr6.sin6_addr = in6addr_any;
+        memcpy(&listenAddr, &listenAddr6, sizeof(listenAddr6));
```

Comment:
Recommend using sizeof(dst) instead of sizeof(src) to avoid potential stack overwrites:
```suggestion
        memcpy(&listenAddr, &listenAddr6, sizeof(listenAddr));
```
Alternatively, choose whichever is smaller:
```suggestion
        memcpy(&listenAddr, &listenAddr6, std::min(sizeof(listenAddr6), sizeof(listenAddr));
```